pygenn/genn_groups.py

```diff
@@ -747,19 +759,19 @@ def push_connectivity_to_device(self):
         self._model.push_connectivity_to_device(self.name)
 
     def load(self):
-        # If synapse population has non-dense connectivity 
+        # If synapse population has non-dense connectivity
         # which requires initialising manually
-        if not self.is_dense and self.is_connectivity_init_required:
-            # If data is available
-            if self.connections_set:
-                if self.is_ragged:
-                    # Get pointers to ragged data structure members
-                    ind = self._assign_ext_ptr_array("ind",
-                                                     self.weight_update_var_size,
-                                                     "unsigned int")
-                    row_length = self._assign_ext_ptr_array("rowLength",
-                                                            self.src.size,
-                                                            "unsigned int")
+        if not self.is_dense and self.weight_sharing_master is None:
+            if self.is_ragged:
+                # Get pointers to ragged data structure members
+                ind = self._assign_ext_ptr_array("ind",
+                                                 self.weight_update_var_size,
+                                                 "unsigned int")
+                row_length = self._assign_ext_ptr_array("rowLength",
+                                                        self.src.size,
+                                                        "unsigned int")
+                # If data is available
```

Comment:
you should also get the max row length GeNN is using with:
```python
max_row_length = self._assign_ext_ptr_single("maxRowLength", "unsigned int")
```
Otherwise, you won't be able to read back variables or connectivity correctly from synapse groups with on-device connectivity

Reply:
does this actually change when something is computed on-device?

Reply:
Actually, forget that, I was totally wrong. ``pop.get_max_row_length()`` is totally fine to use however the connectivity is calculated.